exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +83,531 @@ type BTCMarkets struct {
+		&resp)
+}
+
+// GetTradeHistory returns trade history
+func (b *BTCMarkets) GetTradeHistory(marketID, orderID string, before, after, limit int64) ([]TradeHistoryData, error) {
+	var resp []TradeHistoryData
+	params := url.Values{}
+	if marketID != "" {
+		params.Set("marketId", marketID)
+	}
+	if orderID != "" {
+		params.Set("orderId", orderID)
+	}
+	if before >= 0 {
+		params.Set("before", strconv.FormatInt(before, 10))
 	}
+	if after >= 0 {
+		params.Set("after", strconv.FormatInt(after, 10))
+	}
+	if limit >= 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsTradeHistory, params),
+		nil,
+		&resp)
+}
 
-	return resp.Orders, nil
+// GetTradeByID returns the singular trade of the ID given
+func (b *BTCMarkets) GetTradeByID(id string) (TradeHistoryData, error) {
+	var resp TradeHistoryData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		btcMarketsTradeHistory+"/"+id,
+		nil,
+		&resp)
 }
 
-// GetOpenOrders returns all open orders
-func (b *BTCMarkets) GetOpenOrders() ([]Order, error) {
-	type marketsResp struct {
-		Response
-	}
+// NewOrder requests a new order and returns an ID
+func (b *BTCMarkets) NewOrder(marketID string, price, amount float64, orderType, side string, triggerPrice,
+	targetAmount float64, timeInForce string, postOnly bool, selfTrade, clientOrderID string) (OrderData, error) {
+	var resp OrderData
 	req := make(map[string]interface{})
-	var resp marketsResp
-	path := fmt.Sprintf("/v2/order/open")
-
-	err := b.SendAuthenticatedRequest(http.MethodGet, path, req, &resp)
-	if err != nil {
-		return nil, err
+	req["marketId"] = marketID
+	req["price"] = strconv.FormatFloat(price, 'f', -1, 64)
+	req["amount"] = strconv.FormatFloat(amount, 'f', -1, 64)
+	req["type"] = orderType
+	req["side"] = side
+	if orderType == stopLimit || orderType == takeProfit || orderType == stop {
+		req["triggerPrice"] = strconv.FormatFloat(triggerPrice, 'f', -1, 64)
 	}
-
-	if !resp.Success {
-		return nil, errors.New(resp.ErrorMessage)
+	if targetAmount > 0 {
+		req["targetAmount"] = strconv.FormatFloat(targetAmount, 'f', -1, 64)
+	}
+	if timeInForce != "" {
+		req["timeInForce"] = timeInForce
+	}
+	req["postOnly"] = postOnly
+	if selfTrade != "" {
+		req["selfTrade"] = selfTrade
+	}
+	if clientOrderID != "" {
+		req["clientOrderID"] = clientOrderID
 	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsOrders, req, &resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetOrders returns current order information on the exchange
+func (b *BTCMarkets) GetOrders(marketID string, before, after, limit int64, status string) ([]OrderData, error) {
+	var resp []OrderData
+	params := url.Values{}
+	if marketID != "" {
+		params.Set("marketId", marketID)
+	}
+	if before >= 0 {
```

Comment:
Please add a check to ensure that only before or after can be set and before should be > 0